dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -454,27 +458,24 @@ public Entry createAndSetDiscourseTopic(Long id) throws CustomWebApplicationExce
 
         // Create title and link to entry
 
-        String entryLink = "https://dockstore.org/";
-        String title = "";
-        if (hostName.contains("staging")) {
-            entryLink = "https://staging.dockstore.org/";
-            title = "Staging ";
-        }
+        final String entryPath;
+        final String sitePath;
         if (entry instanceof BioWorkflow) {
-            title += ((BioWorkflow)(entry)).getWorkflowPath();
-            entryLink += "workflows/";
+            entryPath = ((BioWorkflow)entry).getWorkflowPath();
+            sitePath = "workflows/" + entryPath;
         } else if (entry instanceof Service) {
-            title += ((Service)(entry)).getWorkflowPath();
-            entryLink += "services/";
+            entryPath = ((Service)entry).getWorkflowPath();
+            sitePath = "services/" + entryPath;
         } else if (entry instanceof AppTool) {
-            title += ((AppTool)(entry)).getWorkflowPath();
-            entryLink += "tools/";
+            entryPath = ((AppTool)entry).getWorkflowPath();
+            sitePath = "tools/" + entryPath;
         } else {
-            title += ((Tool)(entry)).getToolPath();
-            entryLink += "tools/";
+            entryPath = ((Tool)entry).getToolPath();
+            sitePath = "tools/" + entryPath;
         }
 
-        entryLink += title;
+        final String title = isProduction ? entryPath : (baseUrl + " " + entryPath);
```

Comment:
This means the title will be, for staging, something like "https://staging.dockstore.org github.com/foo/bar", correct? If so, the old way would look a little nicer ("Staging github.com/foo/bar"), but it's only for internal use, so should be OK.

Reply:
Yup.  My rationale there is since it's for internal use, might as well prepend the entire base url, it works for all the platforms (should we expand discuss support to them) and conveys more information.